framework/src/ics/InitialConditionWarehouse.C

```diff
@@ -43,43 +45,42 @@ InitialConditionWarehouse::addObject(std::shared_ptr<InitialConditionBase> objec
       mooseError("You are trying to set a boundary restricted variable on non-nodal variable. That "
                  "is not allowed.");
 
-    std::map<std::string, std::set<BoundaryID>>::const_iterator iter =
-        _boundary_ics[tid].find(var.name());
+    auto iter = _boundary_ics[tid].find(ic_key);
```

Comment:
```suggestion
    const auto iter = _boundary_ics[tid].find(ic_key);
```

Sorry, that should have been const